### ISSUE TITLE
Theme renders incorrectly in theme editor, but right in HTML file

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2727]
@billreed63 

Unfortunately, I'm not sure I have a good reproduction scenario for this. But, I was trying to change some of the properties on primitive widgets in the iPhone theme, and it didn't seem to be working especially well. So, I saved and reopened, and things look  messed up in the theme editor:

![screenshot](http://i.imgur.com/ukxA6.png)

However, the HTML file using the theme still looks correct:

![screenshot](http://i.imgur.com/HBPN7.png)

**Comment by billreed63** [comment]
The top picture is the CSS files are not loaded, where they in the head of the document? If you close and reopen the file do they appear?

**Comment by aerwin** [comment]
What document do you want me to look at?

No, closing and reopening does not fix it.

**Comment by billreed63** [comment]
look at the in the head of the iframe tab that the themeEditor is in Look for the theme in the head you should see something like

```
<head>
        <base href="http://lab9.austin.ibm.com:50010/maqetta/user/billreed/ws/workspace/project1/themes/custom/dojo-theme-editor.html">
        <meta charset="utf-8">
        <script src="/maqetta/app/static/lib/dojo/1.8/dojo/dojo.js" data-dojo-config="'parseOnLoad':false,'async':false,'themeMap':[['.*',null,[]]],'mblThemeFiles':[],'packages':[{'name':'maqettaSamples','location':'../../../samples'},{'name':'maqetta','location':'../../maqetta'},{'name':'dojo-src','location':'..'},{'name':'gridx','location':'../gridx'},{'name':'clipart','location':'../../clipart'},{'name':'shapes','location':'../../shapes'},{'name':'widgets','location':'../../widgets'}]" type="text/javascript"></script>
        <script>
            // base dependencies needed for VE
            require([
                'dojo/parser',
                'dojox/html/_base',
                'dojo/domReady!'
                ,'dijit/dijit','dijit/dijit-all'
            ], top.loading_edit_context_1);
        </script>


    <link rel="stylesheet" type="text/css" href="/maqetta/app/static/lib/dojo/1.8/dijit/themes/claro/claro.css">
       <link rel="stylesheet" type="text/css" href="../iphone1/iphone1.css"><link rel="stylesheet" type="text/css" href="http://lab9.austin.ibm.com:50010/maqetta/user/billreed/ws/workspace/project1/themes/custom/document.css">
       <link rel="stylesheet" type="text/css" href="http://lab9.austin.ibm.com:50010/maqetta/app/davinci/ve/resources/content.css?9d7b2891f0c33f19550c5f403fe1264a14d42de7">
</head>
```

In this example the cloned theme is iphone1.css

strange that closing and reopen does not fix, Can you download your workspace and send to me

**Comment by aerwin** [comment]
Unfortunately, when I tried to download my workspace, it failed (see ##2733). I'll ST you with my password to that server.

**Comment by billreed63** [comment]
This is a timing issue the theme CSS files are not getting added to the head, the only seems to happen on lab9 so it may be the slow network connection. But something in the way we load the html and then add the theme css to the head

@aerwin was clone of iphone here is the delta CSS

```
@import url("../iphone/iphone.css");.mobile body {
    background-color : red;
    background-image : -webkit-gradient(linear, left top, left bottom, from(white), to(black), color-stop(100%, #200f0f));
    background-image : -o-linear-gradient(white, #200f0f 100%, black);
    background-image : -ms-linear-gradient(white, #200f0f 100%, black);
    background-image : -moz-linear-gradient(white, #200f0f 100%, black);
    background-image : -webkit-linear-gradient(white, #200f0f 100%, black);
    background-image : linear-gradient(white, #200f0f 100%, black);
    font-family : 'Courier New',Courier,monospace;
    font-size : 21px;
    font-weight : bold;
    color : green;
}
.mblView {
    color : green;
}
.mblHeading {
    font-family : 'Helvetica,Trebuchet MS',sans-serif;
    font-weight : normal;
    font-style : italic;
    color : blue;
    font-size : 20px;
    background-image : -webkit-gradient(linear, left top, left bottom, from(#8f9ea9), to(green), color-stop(40%, #b7c0c7));
    background-image : -o-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -ms-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -moz-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -webkit-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : linear-gradient(#8f9ea9, #b7c0c7 40%, green);
}
.mblColorDefault {
    font-family : 'Helvetica,Trebuchet MS',sans-serif;
    font-weight : normal;
    font-style : italic;
    color : blue;
    font-size : 20px;
    background-image : -webkit-gradient(linear, left top, left bottom, from(#8f9ea9), to(green), color-stop(40%, #b7c0c7));
    background-image : -o-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -ms-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -moz-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -webkit-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : linear-gradient(#8f9ea9, #b7c0c7 40%, green);
}
.mblEdgeToEdgeCategory {
    font-family : 'Helvetica,Trebuchet MS',sans-serif;
    font-weight : normal;
    font-style : italic;
    color : blue;
    font-size : 20px;
    background-image : -webkit-gradient(linear, left top, left bottom, from(#8f9ea9), to(green), color-stop(40%, #b7c0c7));
    background-image : -o-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -ms-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -moz-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : -webkit-linear-gradient(#8f9ea9, #b7c0c7 40%, green);
    background-image : linear-gradient(#8f9ea9, #b7c0c7 40%, green);
}
.mblToolBarButton {
    background-image : -webkit-gradient(linear, left top, left bottom, from(#cecece), to(green));
    background-image : -o-linear-gradient(#cecece, green);
    background-image : -ms-linear-gradient(#cecece, green);
    background-image : -moz-linear-gradient(#cecece, green);
    background-image : -webkit-linear-gradient(#cecece, green);
    background-image : linear-gradient(#cecece, green);
    color : red;
    font-weight : normal;
    font-family : 'Courier New',Courier,monospace;
}
.mblButton {
    background-image : -webkit-gradient(linear, left top, left bottom, from(#cecece), to(green));
    background-image : -o-linear-gradient(#cecece, green);
    background-image : -ms-linear-gradient(#cecece, green);
    background-image : -moz-linear-gradient(#cecece, green);
    background-image : -webkit-linear-gradient(#cecece, green);
    background-image : linear-gradient(#cecece, green);
    color : red;
    font-weight : normal;
    font-family : 'Courier New',Courier,monospace;
}
.mblToggleButton {
    background-image : -webkit-gradient(linear, left top, left bottom, from(#cecece), to(green));
    background-image : -o-linear-gradient(#cecece, green);
    background-image : -ms-linear-gradient(#cecece, green);
    background-image : -moz-linear-gradient(#cecece, green);
    background-image : -webkit-linear-gradient(#cecece, green);
    background-image : linear-gradient(#cecece, green);
    color : red;
    font-weight : normal;
    font-family : 'Courier New',Courier,monospace;
}
.mblCheckBox {
    background-image : -webkit-gradient(linear, left top, left bottom, from(#cecece), to(green));
    background-image : -o-linear-gradient(#cecece, green);
    background-image : -ms-linear-gradient(#cecece, green);
    background-image : -moz-linear-gradient(#cecece, green);
    background-image : -webkit-linear-gradient(#cecece, green);
    background-image : linear-gradient(#cecece, green);
    color : red;
    font-weight : normal;
    font-family : 'Courier New',Courier,monospace;
}
.mblRadioButton {
    background-image : -webkit-gradient(linear, left top, left bottom, from(#cecece), to(green));
    background-image : -o-linear-gradient(#cecece, green);
    background-image : -ms-linear-gradient(#cecece, green);
    background-image : -moz-linear-gradient(#cecece, green);
    background-image : -webkit-linear-gradient(#cecece, green);
    background-image : linear-gradient(#cecece, green);
    color : red;
    font-weight : normal;
    font-family : 'Courier New',Courier,monospace;
}
.mblSwitchBg {
    background-image : -webkit-gradient(linear, left top, left bottom, from(#cecece), to(green));
    background-image : -o-linear-gradient(#cecece, green);
    background-image : -ms-linear-gradient(#cecece, green);
    background-image : -moz-linear-gradient(#cecece, green);
    background-image : -webkit-linear-gradient(#cecece, green);
    background-image : linear-gradient(#cecece, green);
    color : red;
    font-weight : normal;
    font-family : 'Courier New',Courier,monospace;
}
.mblButtonSelected {
    background-image : none;
    background-color : blue;
    font-weight : bold;
}
```

**Comment by billreed63** [comment]
Hopefully the above fix will get it. The helper had a method to getHeadImports that returned ""  And the helper was called in 

```
if (helper && helper.getHeadImports) {
} else {
 add the links to the head
}
```

So depending on how fast the helpers was loading (async) above this code the helper might or might not have the method, 
I think in the cases where the CSS was added to the head the deferred had not completed yet so the helper which was really a deferred object did not have the method getHeadImports so the link was being added to the head. But some times the helper would have loaded and in that case the method was present thus skipping the else that adds the css to the head.... That is my story and I am sticking to it!

**Comment by billreed63** [comment]
Fixed by  abdf451bfc3602a62aed117278881de1658668aa
